protocol/serialization-types/src/type_annotation.rs

```diff
@@ -0,0 +1,16 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0
+
+/// This trait annotates a given type its different serialization types,
+/// and provide utility functions to easily convert between them
+pub trait TypeAnnotation: Sized {
```

Comment:
Can we name this trait to `DataFormat`, like how they mention of data formats in serde: https://docs.rs/serde/latest/serde/#data-formats

Reply:
I suggest we rename the trait to `SerializationTypes`

Reply:
and the associated types to just `BinProt` and `Json`

Reply:
fixed